api/external/infra_proxy/request/databags.proto

```diff
@@ -39,6 +41,19 @@ message CreateDataBagItem {
     string server_id = 2;
     // Data bag name.
     string name = 3;
-    // Data bag item stringify data.
-    string data = 4;
+    // Data bag item JSON data.
+    google.protobuf.Struct data = 4;
+}
+
+message UpdateDataBagItem {
+    // Chef organization ID.
+    string org_id = 1;
+    // Chef Infra Server ID.
+    string server_id = 2;
+    // Data bag name.
+    string name = 3;
+    // Data bag item ID.
+    string item_id = 4;
+    // Data bag item JSON data.
+    google.protobuf.Struct data = 5;
```

Comment:
Changed data proto3 type to `google.protobuf.Struct` for dynamic JSON attributes.